libsql-sqlite3/src/vectorIndex.c

```diff
@@ -363,18 +363,19 @@ static struct VectorColumnType VECTOR_COLUMN_TYPES[] = {
 struct VectorParamName {
   const char *zName;
   int tag;
-  int type; // 0 - enum, 1 - integer, 2 - float
+  int type; // 0 - string enum, 1 - integer, 2 - float
   const char *zValueStr;
   u64 value;
 };
 
 static struct VectorParamName VECTOR_PARAM_NAMES[] = {
-  { "type",     VECTOR_INDEX_TYPE_PARAM_ID, 0, "diskann", VECTOR_INDEX_TYPE_DISKANN },
-  { "metric",   VECTOR_METRIC_TYPE_PARAM_ID, 0, "cosine", VECTOR_METRIC_TYPE_COS },
-  { "metric",   VECTOR_METRIC_TYPE_PARAM_ID, 0, "l2",     VECTOR_METRIC_TYPE_L2 },
-  { "alpha",    VECTOR_PRUNING_ALPHA_PARAM_ID, 2, 0, 0 },
-  { "search_l", VECTOR_SEARCH_L_PARAM_ID, 1, 0, 0 },
-  { "insert_l", VECTOR_INSERT_L_PARAM_ID, 2, 0, 0 },
+  { "type",      VECTOR_INDEX_TYPE_PARAM_ID,    0, "diskann", VECTOR_INDEX_TYPE_DISKANN },
+  { "metric",    VECTOR_METRIC_TYPE_PARAM_ID,   0, "cosine", VECTOR_METRIC_TYPE_COS },
+  { "metric",    VECTOR_METRIC_TYPE_PARAM_ID,   0, "l2",     VECTOR_METRIC_TYPE_L2 },
+  { "alpha",     VECTOR_PRUNING_ALPHA_PARAM_ID, 2, 0, 0 },
+  { "search_l",  VECTOR_SEARCH_L_PARAM_ID,      1, 0, 0 },
+  { "insert_l",  VECTOR_INSERT_L_PARAM_ID,      1, 0, 0 },
+  { "max_edges", VECTOR_MAX_EDGES_PARAM_ID,     1, 0, 0 },
```

Comment:
`max_edges` seems a bit odd name for the parameter. Did you consider `max_neighbors`?

Reply:
Yes, agreed. The only reason why I choose `edge` here is because I decided to use this terminology in the `diskann` code (because node and edge is more concise "terms" compared to node and neighbor, from my subjective point of view).

But for users we don't need to preserve this internal terminology.

Reply:
Hm....actually, should we choose something else?
I realized that there are two different spellings of the word: `neighbors` and `neighbours` which may be confusing...

Reply:
We can shorten setting name down to `max_neigh`...

Reply:
Or we can support both variants actually